packages/web-sdk/src/instrumentations/webVitalsWithAttribution/instrumentation.ts

```diff
@@ -0,0 +1,128 @@
+import { onCLS, onFCP, onFID, onINP, onLCP, onTTFB } from 'web-vitals/attribution'
+import type { Metric } from 'web-vitals/attribution'
+
+import { BaseInstrumentation, VERSION } from '@grafana/faro-core';
+import type  { MeasurementEvent, PushMeasurementOptions } from '@grafana/faro-core'
+
+type Values = MeasurementEvent['values']
+type Context = Required<PushMeasurementOptions>['context']
+
+export class WebVitalsWithAttributionInstrumentation extends BaseInstrumentation {
+  readonly name = '@grafana/faro-web-sdk:instrumentation-web-vitals-with-attribution'
+  readonly version = VERSION
+
+  initialize(): void {
+    this.logDebug('Initializing');
+    this.measureCLS()
+    this.measureFCP()
+    this.measureFID()
+    this.measureINP()
+    this.measureLCP()
+    this.measureTTFB()
+  }
+
+  private measureCLS(): void {
+    onCLS((metric) => {
+      const values = this.buildInitialValues(metric)
+      this.addIfPresent(values, 'largest_shift_entry', metric.attribution.largestShiftEntry?.value)
+      this.addIfPresent(values, 'largest_shift_value', metric.attribution.largestShiftValue)
+      this.addIfPresent(values, 'largest_shift_time', metric.attribution.largestShiftTime)
+
+      const context = this.buildInitialContext(metric)
+      this.addIfPresent(context, 'load_state', metric.attribution.loadState)
+      this.addIfPresent(context, 'largest_shift_target', metric.attribution.largestShiftTarget)
+
+      this.pushMeasurement(values, context)
+    })
+  }
+
+  private measureFCP(): void {
+    onFCP((metric) => {
```

Comment:
I don't see a problem, I made the change here: https://github.com/grafana/faro-web-sdk/pull/595/commits/b71129dad52ed3432d1b6a7d2e2c3d7d42d7609e

Is it relevant to have this mapping across all metrics? Or is it better to just map the TTFB, FCP and LCP metrics?

I don't like the idea of ​​web-vitals instrumentation depending on performance instrumentation with the `NAVIGATION_ID_STORAGE_KEY` import.

I thought about raising this const to a higher level but I preferred to validate with you first

Reply:
> Is it relevant to have this mapping across all metrics? Or is it better to just map the TTFB, FCP and LCP metrics?

I think it's fine to map across metrics. This gives an option to scope them to a specific navigation. Might be helpful.

> I don't like the idea of ​​web-vitals instrumentation depending on performance instrumentation with the NAVIGATION_ID_STORAGE_KEY import.
I thought about raising this const to a higher level but I preferred to validate with you first

Yes sounds good to me. Like raising it up one level to a `instrumentationConstants` file.
This will also act as hint for other devs that the const is used in other instruments as well. 
Go for it 👍

Reply:
> I think it's fine to map across metrics. This gives an option to scope them to a specific navigation. Might be helpful.

OK, I'll leave it as it is then

> Yes sounds good to me. Like raising it up one level to a instrumentationConstants file.
This will also act as hint for other devs that the const is used in other instruments as well.
Go for it 👍

Great, I made the change here: https://github.com/grafana/faro-web-sdk/pull/595/commits/df1bbbce597b5c15a3b86040957bdb45013e8949